microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -194,17 +197,17 @@ def _generate_testpmd_command(self, nic_to_include: str) -> str:
         #   --stats-period <display interval in seconds>
         return (
             f"{self._testpmd_install_path} -l 0-1 -n 4 --proc-type=primary "
-            + f"{nic_to_include} -- --forward-mode=txonly -a --stats-period 1"
+            + f"{nic_to_include} -- --forward-mode={mode} -a --stats-period 1"
         )
 
-    def run_with_timeout(self, nic_to_include: str, timeout: int) -> str:
+    def run_with_timeout(self, cmd: str, timeout: int) -> str:
         timer_proc = self.node.execute_async(
-            f"sleep {timeout} && killall -s INT {self._testpmd_install_path}",
+            f"sleep {timeout} && killall -s INT {cmd.split()[0]}",
```

Comment:
Just noticed this logic about timeout. The `wait_result` has a timeout method. Does it work here?

Reply:
I'm not sure if the timeout will wait until it reaches every time. If so, it's low efficient.

Reply:
It does not. Killing the program correctly requires the INT signal in this case.

Reply:
Cab you add a method in process class to `send_signal` to process? There is such a method in internal process classes, but not exposed.
How about the second question about the timeout? Does it make sure the process won't sleep `timeout` seconds every time?

Reply:
> I'm not sure if the timeout will wait until it reaches every time. If so, it's low efficient.

It will wait until 30 seconds each time but that's the desired result, we want to run the program for 30 seconds. I'm realizing the name might be misleading, it's less a timeout and more of 'run_for_n_seconds'

Reply:
30 seconds is pretty long. Can it be tested in shorter time, how about 5 seconds?